### DISCUSSION TITLE
Algolia search response not displaying

### DISCUSSION THREAD

**Discussion by boilemmashem** [facebook/docusaurus#8970] [Q&A] [answered]
Hi all, new here :)

sourcecode: https://github.com/mikehaart/wicbwwiki
production: https://wicbwwiki.netlify.app

I'm using Docusaurus for a hobby project. I'm hosted on Netlify so I've installed the integration but when I use the search function on my site the response doesn't display in the UI. Looks like the request fires correctly and the response sends the right index information but getting nothing on the page.

Not sure what to look into, suggestions welcome!

**Comment by slorber** [comment]
Hey

You are using optional options that you probably shouldn't here:

```js
     algolia: {
        // The application ID provided by Algolia
        appId: 'DGMONGHJKH',
        // Public API key: it is safe to commit it
        apiKey: '211080b0a4c2c33dc0a2d0e25ac0c0f6',
        indexName: 'netlify_2b25d0d1-0404-4bf1-a96d-fe292ed9901b_main_all',
        // Optional: see doc section below
        contextualSearch: false,
        // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
        externalUrlRegex: 'external\\.com|domain\\.com',
        // Optional: Replace parts of the item URLs from Algolia. Useful when using the same search index for multiple deployments using a different baseUrl. You can use regexp or string in the `from` param. For example: localhost:3000 vs myCompany.com/docs
        replaceSearchResultPathname: {
          from: '/docs/', // or as RegExp: /\/docs\//
          to: '/',
        },
        // Optional: Algolia search parameters
        searchParameters: {},
        // Optional: path for search page that enabled by default (`false` to disable it)
        searchPagePath: 'search',
        //... other Algolia params
      }
```

Not sure what is happening but I'd recommend to use a more minimal config to get started:

```js
     algolia: {
        appId: 'DGMONGHJKH',
        apiKey: '211080b0a4c2c33dc0a2d0e25ac0c0f6',
        indexName: 'netlify_2b25d0d1-0404-4bf1-a96d-fe292ed9901b_main_all',
      }
```

Your index name looks weird to me, are you sure it's the Algolia index name?

Also note that you must ensure that the Algolia crawler did run (by default once every 24h but you can trigger it manually). The Algolia dashboard can tell if your index contains records or not, so make sure to double-check that your index is not empty.

**Reply by boilemmashem** [reply]
Thanks, ive changed the config as you suggested.

My index name was created automatically when I installed the Algolia crawler integration in Netlify. The index itself looks good since I can search through it on the Algolia dashboard

![Screenshot 2023-05-11 at 16 32 08](https://github.com/facebook/docusaurus/assets/17619758/500a7114-ba94-480f-903e-a37ff6666892)

When try to search on my production site, I can see the search response coming back in my browser devtools but it's not loading onto the page.

**Reply by slorber** [reply]
Note for self: Netlify deploy links: https://app.netlify.com/sites/wicbwwiki/deploys

Not sure what is happening here. @shortcuts any idea?

Did you run the crawler again since the last deployment? Are you sure you are using a valid Docusaurus crawler config?

**Reply by slorber** [reply]
Please give me the exact search query you are currently using because right now all queries I try return 0 hits. 
(Was able to get some hits yesterday)

For example, no hits (even in network devtools) using this query:
https://wicbwwiki.netlify.app/search?q=Arryn+Quinn

I believe it could be fixed by reindexing, now that contextualSearch has been turned on again.

But before running the crawler make sure you use the recommended crawler config: https://docsearch.algolia.com/docs/templates/#docusaurus-v2-template

If you decide to change this config, it's your responsibility to make it work

**Reply by boilemmashem** [reply]
Thanks for your help!

I reran the crawler to make a fresh index and seem to be betting an error now when I search anything. The search term I've been using is "sm". I'm getting hits from that search but also getting an error now.

```
can't access property "lvl1", item.hierarchy is undefined
getItems/<@webpack-internal:///./node_modules/@docsearch/react/dist/esm/DocSearchModal.js:263:19
groupBy/<@webpack-internal:///./node_modules/@docsearch/react/dist/esm/utils/groupBy.js:7:24
groupBy@webpack-internal:///./node_modules/@docsearch/react/dist/esm/utils/groupBy.js:6:17
getItems@webpack-internal:///./node_modules/@docsearch/react/dist/esm/DocSearchModal.js:262:85
onInput/request</<@webpack-internal:///./node_modules/@algolia/autocomplete-core/dist/esm/onInput.js:89:37
onInput/request<@webpack-internal:///./node_modules/@algolia/autocomplete-core/dist/esm/onInput.js:88:32
```

I cant seem to find the config options for Algolia crawler anywhere - maybe due to it being created via Netlify integration 🤷 

Could it be a formatting issue with my crawler?

**Reply by slorber** [reply]
I have no idea what is going on 😅 

@shortcuts could it be related to the "skip to content" change you did recently?

It looks like the search hits do not have a hierarchy for this site, and it's causing problems, and there's a weird mention of "skip to content" in search hits

![CleanShot 2023-05-15 at 12 43 48@2x](https://github.com/facebook/docusaurus/assets/749374/a4df354f-7bb2-4df4-b80e-57d4f995f960)

---

> I cant seem to find the config options for Algolia crawler anywhere - maybe due to it being created via Netlify integration 🤷

Yes agree the crawler config is not easy to find in Algolia UI 😅 

Go there: https://crawler.algolia.com/admin

**Reply by boilemmashem** [reply]
Had another look for the crawler config at https://crawler.algolia.com/admin but theres no such option. 

I found a thread in the algolia community forums with the same issue https://discourse.algolia.com/t/docusaurus-v2-netlify-index-without-results/17662 Seems the poster of that thread also cant find the config. I've made [a post there](https://discourse.algolia.com/t/netlify-plugin-crawler-config-docusaurus-v2/18112) to see if I can get some guidance on accessing it.

**Comment by boilemmashem** [comment, marked as answer]
It seems that when you use the Netlify plugin to integrate with Agolia search you are [unable to access the crawler config](https://discourse.algolia.com/t/netlify-plugin-crawler-config-docusaurus-v2/18112/5?u=michael-hart). This presents a problem for Docusaurus since it expects the index to be presented in a certain way. As far as I can see the solution at the moment is just to simply avoid using that plugin.

**Reply by shortcuts** [reply]
Hey @mikehaart @slorber, sorry I was in vacation :) 

Indeed, the Netlify x Algolia Crawler integration doesn't natively support DocSearch, and there's no way for you to update the Crawler configuration to make it work properly. A potential workaround would be to use [`transformItems`](https://docsearch.algolia.com/docs/api/#transformitems) so that you can transform the API response to make it match the expected DocSearch UI record shape. 

If your project is open-source, just apply there https://docsearch.algolia.com/apply and you'll get a fully working and configurable crawler in no time! :D

**Reply by boilemmashem** [reply]
Thanks for the follow up :)
My project is open source but not technical documentation, just a hobby project. 

Do you know of any examples of the `transformItems` api call in situe? I'm new to handling requests and would love to know more working on this.
